flocker/node/agents/testtools/_blockdevice.py

```diff
@@ -869,3 +881,237 @@ def setUp(self):
+    api = get_api(
+        backend=backend,
+        api_args=api_args,
+        reactor=reactor,
+        cluster_id=make_cluster_id(
+            # XXX I assume that this function will only be used from functional
+            # tests.  Acceptance tests will load the API via the AgentService.
+            test_type=TestTypes.FUNCTIONAL,
+            # XXX This should match the backend, but Providers are
+            # ValueConstants rather than NamedConstants so we can't
+            # look them up and I'm not convinced whether they add any
+            # value.
+            provider=Providers.UNSPECIFIED,
+        ),
+    )
+    return api
+
+
+CONFIG_OVERRIDE_PREFIX = 'FLOCKER_FUNCTIONAL_TEST_CONFIG_OVERRIDE_'
+
+
+def get_blockdevice_config():
+    """
+    Get configuration dictionary suitable for use in the instantiation
+    of an ``IBlockDeviceAPI`` implementation.
+
+    :raises: ``InvalidConfig`` if a
+        ``FLOCKER_FUNCTIONAL_TEST_CLOUD_CONFIG_FILE`` was not set and the
+        default config file could not be read.
+
+    :return: XXX
+    """
+    flocker_functional_test = environ.get('FLOCKER_FUNCTIONAL_TEST')
+    if flocker_functional_test is None:
+        raise SkipTest(
+            'Please set FLOCKER_FUNCTIONAL_TEST environment variable to '
+            'run storage backend functional tests.'
+        )
+
+    config_file_path = environ.get('FLOCKER_FUNCTIONAL_TEST_CLOUD_CONFIG_FILE')
+    if config_file_path is None:
+        raise SkipTest(
+            'Supply the path to a backend configuration file '
+            'using the FLOCKER_FUNCTIONAL_TEST_CLOUD_CONFIG_FILE environment '
+            'variable.'
+        )
+
+    # ie storage-drivers.rackspace
+    config_section = environ.get(
+        'FLOCKER_FUNCTIONAL_TEST_CLOUD_CONFIG_SECTION',
+    )
+    if config_section is None:
+        raise SkipTest(
+            'Supply the section of the config file '
+            'containing the configuration for the driver under test '
+            'with the FLOCKER_FUNCTIONAL_TEST_CLOUD_CONFIG_SECTION '
+            'environment variable.'
+        )
+
+    with open(config_file_path) as config_file:
+        config = yaml.safe_load(config_file.read())
+
+    section = None
+    for section in config_section.split('.'):
+        config = config.get(section)
+
+    if config is None:
+        raise InvalidConfig(
+            "The requested section "
+            "was not found in the configuration file. "
+            "Section: %s, "
+            "Configuration File: %s" % (config_section, config_file_path)
+        )
+
+    # XXX A hack to work around the fact that the sub-sections of
+    # storage-drivers in acceptance.yml do not all have a ``backend`` key.
+    if "backend" not in config:
+        config["backend"] = section
+
+    # XXX Allow certain configuration to be overridden from environment
+    # variables.
+    # This allows us to dynamically use the zone where the functional test
```

Comment:
I tried  parsing the environment variable value as JSON or YAML but it seems ....error prone. Especially when I'm trying to insert multiple values and values expanded from bash variables.
I'll leave the current mechanism for now (undocumented) and hopefully we can remove it as soon as we update all the drivers to auto-detect their zone.